### PR TITLE
utils: format path for listing files

### PR DESCRIPTION
**Pull request by aravind-wdc**
format the path provided by user, to list files.

Signed-off-by: Aravind Ramesh <aravind.ramesh@wdc.com>

### CODE REVIEW

**Comment by aravind-wdc** [comment]
Fixes issue reported by https://github.com/westerndigitalcorporation/zenfs/issues/30

**Review by yhr — CHANGES_REQUESTED** [review]
The commit message does not provide a description of 1) what the issue is and 2) how this patch fixes it.

Also describe any other changes you have done (like returning an error if the directory does not exist)

File: util/zenfs.cc [hunk 1]
```diff
@@ -142,6 +145,23 @@ void list_children(ZenFS *zenFS, std::string path) {
...
+{
+  std::size_t pos = path.find('/', 0);
+  while (pos != std::string::npos) {
+    while (path.compare(pos + 1, 1, "/") == 0) {
+      path.erase(pos + 1, 1);
+    }
+    pos = path.find('/', pos + 1);
+  }
+
+  if (path.front() == '/')
```
Comment (skyzh) [inline util/zenfs.cc:158]: `path.front()` and `path.back()` would return a dangling reference if the string is empty at this point. Shall we add something like
```
if (path.empty()) {
  path = "/";
}
```

here?

Comment (skyzh) [reply util/zenfs.cc:158]: It turns out the C++ standard guarantees there would be a `\0` if string is empty. No issue here.

**Comment by aravind-wdc** [comment]
$ sudo ./zenfs list --zbd=nvme0n1 --path=rocksdbtest/dbbench
           0    Aug 16 2021 23:07:43            LOCK                            
       39116    Aug 16 2021 23:07:43            LOG                             
   316552720    Aug 16 2021 23:07:43            000012.sst                      
   217338191    Aug 16 2021 23:07:43            000013.log                      
   194483993    Aug 16 2021 23:07:43            000014.sst                      
          16    Aug 16 2021 23:07:43            CURRENT                         
          37    Aug 16 2021 23:07:43            IDENTITY                        
           0    Aug 16 2021 23:07:43            LOCK                            
       39116    Aug 16 2021 23:07:43            LOG                             
         584    Aug 16 2021 23:07:43            MANIFEST-000004                 
        6317    Aug 16 2021 23:07:43            OPTIONS-000007 

$ sudo ./zenfs list --zbd=nvme0n1 --path=rocksdbtest/dbbench/
           0    Aug 16 2021 23:07:43            LOCK                            
       39116    Aug 16 2021 23:07:43            LOG                             
   316552720    Aug 16 2021 23:07:43            000012.sst                      
   217338191    Aug 16 2021 23:07:43            000013.log                      
   194483993    Aug 16 2021 23:07:43            000014.sst                      
          16    Aug 16 2021 23:07:43            CURRENT                         
          37    Aug 16 2021 23:07:43            IDENTITY                        
           0    Aug 16 2021 23:07:43            LOCK                            
       39116    Aug 16 2021 23:07:43            LOG                             
         584    Aug 16 2021 23:07:43            MANIFEST-000004                 
        6317    Aug 16 2021 23:07:43            OPTIONS-000007    
    
$ sudo ./zenfs list --zbd=nvme0n1 --path=rocksdbtest/dbbench/////
           0    Aug 16 2021 23:07:43            LOCK                            
       39116    Aug 16 2021 23:07:43            LOG                             
   316552720    Aug 16 2021 23:07:43            000012.sst                      
   217338191    Aug 16 2021 23:07:43            000013.log                      
   194483993    Aug 16 2021 23:07:43            000014.sst                      
          16    Aug 16 2021 23:07:43            CURRENT                         
          37    Aug 16 2021 23:07:43            IDENTITY                        
           0    Aug 16 2021 23:07:43            LOCK                            
       39116    Aug 16 2021 23:07:43            LOG                             
         584    Aug 16 2021 23:07:43            MANIFEST-000004                 
        6317    Aug 16 2021 23:07:43            OPTIONS-000007

$ sudo ./zenfs list --zbd=nvme0n1 --path=///rocksdbtest/dbbench/////
           0    Aug 16 2021 23:07:43            LOCK                            
       39116    Aug 16 2021 23:07:43            LOG                             
   316552720    Aug 16 2021 23:07:43            000012.sst                      
   217338191    Aug 16 2021 23:07:43            000013.log                      
   194483993    Aug 16 2021 23:07:43            000014.sst                      
          16    Aug 16 2021 23:07:43            CURRENT                         
          37    Aug 16 2021 23:07:43            IDENTITY                        
           0    Aug 16 2021 23:07:43            LOCK                            
       39116    Aug 16 2021 23:07:43            LOG                             
         584    Aug 16 2021 23:07:43            MANIFEST-000004                 
        6317    Aug 16 2021 23:07:43            OPTIONS-000007      

$ sudo ./zenfs list --zbd=nvme0n1 --path=///rocksdbtest/dbb/
Error: NotFound:  rocksdbtest/dbb/
$

**Review by yhr — APPROVED** [review]
Great, now i understand what was changed, thanks!

**Pull request merged**